sdk/helper/clientcountutil/clientcountutil.go

```diff
@@ -0,0 +1,383 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+// Package clientcountutil provides a library to generate activity log data for
+// testing.
+package clientcountutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/sdk/helper/clientcountutil/generation"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type ActivityLogDataGenerator struct {
+	data            *generation.ActivityLogMockInput
+	addingToMonth   *generation.Data
+	addingToSegment *generation.Segment
+	client          *api.Client
+}
+
+// NewActivityLogData creates a new instance of an activity log data generator
+// The type returned by this function cannot be called concurrently
+func NewActivityLogData(client *api.Client) *ActivityLogDataGenerator {
+	return &ActivityLogDataGenerator{
+		client: client,
+		data:   new(generation.ActivityLogMockInput),
+	}
+}
+
+// NewCurrentMonthData opens a new month of data for the current month. All
+// clients will continue to be added to this month until a new month is created
+// with NewMonthDataMonthsAgo.
+func (d *ActivityLogDataGenerator) NewCurrentMonthData() *ActivityLogDataGenerator {
+	d.newMonth(&generation.Data{Month: &generation.Data_CurrentMonth{CurrentMonth: true}})
+	return d
+}
+
+// NewMonthDataMonthsAgo opens a new month of data, where the clients will be
+// recorded as having been seen monthsAgo months ago. All clients will continue
+// to be added to this month until a new month is created with
+// NewMonthDataMonthsAgo or NewCurrentMonthData.
+func (d *ActivityLogDataGenerator) NewMonthDataMonthsAgo(monthsAgo int) *ActivityLogDataGenerator {
```

Comment:
How do you feel about `NewPreviousMonthData` instead? It would keep the naming scheme consistent and we already have `monthsAgo` context provided by the argument.

Reply:
+1